wranglertools/import_data.py

```diff
@@ -154,65 +153,76 @@ def md5(path):
+    """
     ftp_attach = False
     if not os.path.isfile(path):
         # if the path does not exist, check if it works as a URL
         if path.startswith("ftp://"):  # grab the file from ftp
             print("\nINFO: Attempting to download file from this url %s" % path)
-            with closing(urllib2.urlopen(path)) as r:
-                file_name = path.split("/")[-1]
-                with open(file_name, 'wb') as f:
-                    shutil.copyfileobj(r, f)
-                    path = file_name
-                    ftp_attach = True
+            try:
+                # import pdb; pdb.set_trace()
+                with closing(urllib2.urlopen(path)) as r:
+                    file_name = path.split("/")[-1]
+                    with open(file_name, 'wb') as f:
+                        shutil.copyfileobj(r, f)
+                        path = file_name
+                        ftp_attach = True
+            except urllib2.URLError as e:
+                raise Exception("\nERROR : FTP fetch for 'attachment' failed - {}".format(e))
         else:
             try:
                 r = requests.get(path)
-            except requests.exceptions.MissingSchema:
-                print("\nERROR : The 'attachment' field contains INVALID FILE PATH or URL ({})\n".format(path))
-                sys.exit(1)
-            # if it works as a URL, but does not return 200
-            if r.status_code is not 200:  # pragma: no cover
-                print("\nERROR : The 'attachment' field contains INVALID URL ({})\n".format(path))
-                sys.exit(1)
+            except Exception:
+                raise Exception("\nERROR : The 'attachment' field has INVALID FILE PATH or URL ({})\n".format(path))
+            else:
+                # if it works as a URL, but does not return 200
+                if r.status_code is not 200:  # pragma: no cover
+                    raise Exception("\nERROR : The 'attachment' field has INVALID URL ({})\n".format(path))
             # parse response
             path = path.split("/")[-1]
-            with open(path, "wb") as outfile:
-                outfile.write(r.content)
-
-    filename = os.path.basename(path)
-    mime_type = mimetypes.guess_type(path)[0]
-    major, minor = mime_type.split('/')
-    detected_type = magic.from_file(path, mime=True)
-    # XXX This validation logic should move server-side.
-    if not (detected_type == mime_type or
-            detected_type == 'text/plain' and major == 'text'):
-        if not (minor == 'zip' or major == 'text'):  # zip files are special beasts
-            raise ValueError('Wrong extension for %s: %s' % (detected_type, filename))
+            try:
+                with open(path, "wb") as outfile:
+                    outfile.write(r.content)
+                    ftp_attach = True
+            except Exception as e:
+                raise Exception("\nERROR : Cannot write a tmp file to disk - {}".format(e))
+
+    allowed_mimes = (
```

Comment:
I'd have this be a top level constant instead of a local variable

Reply:
compromised to make it a constant within the function